src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -690,61 +686,62 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 			}
 		}
 
-		// Convert to typed editors and options
-		const typedEditors: IEditorInputWithOptions[] = editors.map(editor => {
-			if (isEditorInputWithOptions(editor)) {
-				return editor;
-			}
-
-			return {
-				editor: this.createEditorInput(editor),
-				options: editor.options
-			};
-		});
-
 		// Find target groups to open
-		const mapGroupToEditorsCandidates = new Map<IEditorGroup, IEditorInputWithOptions[]>();
+		const mapGroupToEditorsCandidates = new Map<IEditorGroup, Array<IEditorInputWithOptions | IUntypedEditorInput>>();
 		if (group === SIDE_GROUP) {
-			mapGroupToEditorsCandidates.set(this.findSideBySideGroup(), typedEditors);
+			mapGroupToEditorsCandidates.set(this.findSideBySideGroup(), editors);
 		} else {
-			for (const typedEditor of typedEditors) {
-				const targetGroup = this.findTargetGroup(typedEditor.editor, typedEditor.options, group);
+			for (const editor of editors) {
+
+				const targetGroup = isEditorInputWithOptions(editor) ? this.findTargetGroup(editor.editor, editor.options, group) : this.findTargetGroup(editor, editor.options, group);
 
 				let targetGroupEditors = mapGroupToEditorsCandidates.get(targetGroup);
 				if (!targetGroupEditors) {
 					targetGroupEditors = [];
 					mapGroupToEditorsCandidates.set(targetGroup, targetGroupEditors);
 				}
 
-				targetGroupEditors.push(typedEditor);
+				targetGroupEditors.push(editor);
 			}
 		}
 
 		// Resolve overrides
 		const mapGroupToEditors = new Map<IEditorGroup, IEditorInputWithOptions[]>();
-		for (const [group, editorsWithOptions] of mapGroupToEditorsCandidates) {
-			for (const { editor, options } of editorsWithOptions) {
-				let editorOverride: IEditorInputWithOptionsAndGroup | undefined;
+		for (const [group, editors] of mapGroupToEditorsCandidates) {
+			for (const editor of editors) {
+
+				const targetGroup = group;
+				let targetGroupEditors = mapGroupToEditors.get(targetGroup);
+				if (!targetGroupEditors) {
+					targetGroupEditors = [];
+					mapGroupToEditors.set(targetGroup, targetGroupEditors);
+				}
+
+				let editorOverride: IEditorInputWithOptions | undefined;
+				const options = editor.options;
 				if (options?.override !== EditorOverride.DISABLED) {
-					const overrideResult = await this.editorOverrideService.resolveEditorOverride(editor, options, group);
+					const editorToOverride = isEditorInputWithOptions(editor) ? editor.editor.toUntyped(group.id, UntypedEditorContext.Default) : editor;
```

Comment:
This code is quite involved. I would suggest to make `resolveEditorOverride` accept `IEditorInputWithOptions` to make this easier to read and deprecate that overload so that we know we have to remove it eventually?

Reply:
I think this is ok now, eventully typed editor support will be gone from the editor service methods.